### PR TITLE
Add subdirectory recursion for finding json data files

### PR DESCRIPTION
**Pull request by gartung**
Update piechart.php and cgi-bin/datalist.py to search subdirectories under data for json files.

### CODE REVIEW

**Comment by gartung** [comment]
piechart.php is working so far
https://gartung.web.cern.ch/gartung/circles/web/piechart.php?local=false&dataset=CMSSW_11_2_X_2020-08-25-2300%2Fslc7_amd64_gcc820%2Fstep3_RAW2DIGI_L1Reco_RECO_RECOSIM_PU.resources&resource=time_real&colours=.%2Fdefault&groups=.%2Fhlt&threshold=0

**Comment by gartung** [comment]
works for piechart.html as well now
https://gartung.web.cern.ch/gartung/circles/web/piechart.html?local=false&dataset=CMSSW_11_2_X_2020-08-25-2300%2Fslc7_amd64_gcc820%2Fstep3_RAW2DIGI_L1Reco_RECO_RECOSIM_PU.resources&resource=time_real&colours=default&groups=hlt&threshold=0

File: web/piechart.php [hunk 1]
```diff
@@ -35,10 +35,21 @@
...
       function preformat($file) {
-        return "'" . basename($file, ".json") . "'";
+        $file = explode('/', $file);
+        unset($file[0]);
+        $file = implode('/', $file);
+        return "'" . dirname($file) . "/" . basename($file, ".json") . "'";
       }
 
-      print("var datasets = [ " . join(", ", array_map("preformat", glob("data/*.json"))) . " ];\n");
+      // Does not support flag GLOB_BRACE
```
Comment (fwyzard) [inline web/piechart.php:44]:
```suggestion
      // from https://stackoverflow.com/a/17161106/2050986
      // does not support flag GLOB_BRACE
```

File: web/cgi-bin/datalist.py [hunk 2]
```diff
@@ -1,15 +1,23 @@
 #! /usr/bin/python
 
-import sys, os, os.path, glob
+import sys, os, os.path, glob, fnmatch
 
 print "Content-Type: text/javascript;charset=utf-8\n"
 
+def find_files(directory, pattern):
```
Comment (fwyzard) [inline web/cgi-bin/datalist.py:7]:
```suggestion
// from https://stackoverflow.com/a/2186673/2050986
def find_files(directory, pattern):
```

**Pull request merged**